doc/api/buffer.md

```diff
@@ -1340,6 +1340,10 @@ console.log(b.indexOf('b', null));
 console.log(b.indexOf('b', []));
 ```
 
+If `value` is an empty string or `Buffer` and `byteOffset` is less
```

Comment:
Nit: `empty string or empty Buffer` would avoid confusion over whether or not the `Buffer` must be empty.

Might be even simpler to replace `is an empty string or Buffer` with `is empty` or `has a length of zero`?

> If `value` is empty and `byteOffset` is less

or 

> If `value` has a length of zero and byteOffset is less